APMrover2/GCS_Mavlink.cpp

```diff
@@ -1238,6 +1238,26 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
         break;
     }
 
+    case MAVLINK_MSG_ID_MANUAL_CONTROL:
+    {
+        if (msg->sysid != rover.g.sysid_my_gcs) {  // Only accept control from our gcs
+            break;
+        }
+
+        mavlink_manual_control_t packet;
+        mavlink_msg_manual_control_decode(msg, &packet);
+
+        bool override_active = false;
+        int16_t roll = (packet.y == INT16_MAX) ? 0 : rover.channel_steer->get_radio_min() + (rover.channel_steer->get_radio_max() - rover.channel_steer->get_radio_min()) * (packet.y + 1000) / 2000.0f;
+        int16_t throttle = (packet.z == INT16_MAX) ? 0 : rover.channel_throttle->get_radio_min() + (rover.channel_throttle->get_radio_max() - rover.channel_throttle->get_radio_min()) * (packet.z + 1000) / 2000.0f;
+        override_active |= hal.rcin->set_override(uint8_t(rover.rcmap.roll() - 1), roll);
+        override_active |= hal.rcin->set_override(uint8_t(rover.rcmap.throttle() - 1), throttle);
```

Comment:
If we look at how it is done on RC_OVERRIDE, it is not need. And I agree. If we received an RC_OVERRIDE or MANUAL_CONTROL message, if the message is valide, it means that we have a connection with the GCS even if the message is empty. 
And set_override return true only if the override is > 0 and on a valid channel but override = -1 is valid too ...It think there is some possible improvements in this logic